feat: [CHAOS-925]: Documentation updates for Chaos Module (#504)
harness/developer-hub: docs/chaos-engineering/chaos-faults/kubernetes/node/node-drain.md

---
id: node-drain
title: Node drain
---

> Node drain drains the node of all its resources running on it. 

Reviewer: ```suggestion
Node drain fault drains all the resources running on a node. 
```

- Due to this, services running on the target node should be rescheduled to run on other nodes. 

![Node Drain](./static/images/node-drain.png)

## Usage
<details>
<summary>View the uses of the fault</summary>
<div>

> This fault determines the resilience of the application when the nodes are deprived of resources. 

Reviewer: ```suggestion
This fault determines the resilience of the application when the application replicas scheduled on a node are removed. It validates the application failover capabilities when a node suddenly becomes unavailable.
```
Node Drain has nothing to do with the deprivation of resources.
